pandas/core/groupby/grouper.py

```diff
@@ -875,7 +875,7 @@ def is_in_obj(gpr) -> bool:
             exclusions.add(gpr.name)
 
         elif is_in_axis(gpr):  # df.groupby('name')
-            if gpr in obj:
+            if not isinstance(obj, Series) and gpr in obj:
```

Comment:
Can you use `obj.ndim != 1` instead of the isinstance check.

Reply:
Yes. Thank you for your review.